apps/examples/testcase/ta_tc/wifi_manager/itc/itc_wifi_manager_main.c

```diff
@@ -303,6 +319,436 @@ static void itc_wifi_manager_scan_ap_p(void)
+ * @brief            Initialize  and de-initalization of wifi manager
+ * @scenario
+ * @apicovered
+ * @precondition     none
+ * @postcondition    none
+ */
+
+static void itc_wifi_manager_reinit_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_init(&wifi_callbacks);
+	TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_SUCCESS);
+
+	ret = wifi_manager_init(&wifi_callbacks);
+	TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_INITIALIZED);
+
+	ret = wifi_manager_deinit();
+	TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_SUCCESS);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @testcase         itc_wifi_manager_init_redeinit_p
+ * @brief            Initialize  and de-initalization of wifi manager
+ * @scenario
+ * @apicovered
+ * @precondition     wifi_manager_init
+ * @postcondition    none
+ */
+
+static void itc_wifi_manager_redeinit_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_init(&wifi_callbacks);
+	TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_SUCCESS);
+
+	ret = wifi_manager_deinit();
+	TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_SUCCESS);
+
+	ret = wifi_manager_deinit();
+	TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_DEINITIALIZED);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @testcase         itc_wifi_manager_reconnect_p
+ * @brief            reconnecting AP when already connected
+ * @scenario
+ * @apicovered
+ * @precondition     wifi_manager_connect_ap
+ * @postcondition    wifi_manager_deinit
+ */
+
+static void itc_wifi_manager_reconnect_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_init(&wifi_callbacks);
+	TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_SUCCESS);
+
+	ret = wifi_manager_set_mode(STA_MODE, NULL);
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_SUCCESS);
+
+	wifi_manager_ap_config_s config;
+	config.ssid_length = strlen(CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_SSID);
+	config.passphrase_length = strlen(CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_PASSPHRASE);
+	strncpy(config.ssid, CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_SSID, config.ssid_length + 1);
+	strncpy(config.passphrase, CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_PASSPHRASE, config.passphrase_length + 1);
+	config.ap_auth_type = (wifi_manager_ap_auth_type_e)CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_AUTHENTICATION;
+	config.ap_crypto_type = (wifi_manager_ap_crypto_type_e)CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_CRYPTO;
+	printf("AP config: %s(%d), %s(%d), %d %d\n", config.ssid, config.ssid_length, config.passphrase, \
+			config.passphrase_length, config.ap_auth_type, config.ap_crypto_type);
+
+	WIFITEST_WAIT;
+	sleep(5);
+	ret = wifi_manager_connect_ap(&config);
+	TC_ASSERT_EQ("wifi_manager_connect_ap", ret, WIFI_MANAGER_SUCCESS);
+	WIFITEST_WAIT;
+
+	sleep(5);
+	ret = wifi_manager_connect_ap(&config);
+	TC_ASSERT_EQ("wifi_manager_connect_ap", ret, WIFI_MANAGER_SUCCESS);
```

Comment:
Is returning ok valid even though it reconnects to same AP?

Reply:
As per API "wifi_manager_connect_ap" code snippets , It should return WIFI_MANAGER_SUCCESS. 

Filename : framework/src/wifi_manager/wifi_utils.c" Line : 466
...
ret = WiFiNetworkJoin((uint8_t *)ap_connect_config->ssid, ap_connect_config->ssid_length, NULL, config);
        if (ret != SLSI_STATUS_SUCCESS) {
                if (ret == SLSI_STATUS_ALREADY_CONNECTED) {
                        nvdbg("WiFiNetworkJoin already connected\n");
                        result = WIFI_UTILS_SUCCESS;
                } else {

...